compiler/resolution/virtualDispatch.cpp

```diff
@@ -751,6 +781,155 @@ static void filterVirtualChildren() {
   }
 }
 
+// map from this type -> name -> fns
+typedef std::map<const char*, std::vector<FnSymbol*> > NameToFns;
+typedef std::map<AggregateType*, NameToFns > TypeToNameToFns;
+
+static void findFunctionsProbablyMatching(TypeToNameToFns & map,
+                                          FnSymbol* theFn,
+                                          AggregateType* ct,
+                                          std::vector<FnSymbol*> & matches) {
+
+  // Find sub-map
+  if (map.count(ct) &&
+      map[ct].count(theFn->name)) {
+    std::vector<FnSymbol*> & fns = map[ct][theFn->name];
+
+    // Find functions probably matching here
+    for_vector(FnSymbol, fn, fns) {
+      if (possibleSignatureMatch(theFn, fn))
+        matches.push_back(fn);
+    }
+  }
+
+  // Check parent types
+  forv_Vec(AggregateType, pt, ct->dispatchParents) {
+    findFunctionsProbablyMatching(map, theFn, pt, matches);
+  }
+}
+
+// This function checks that the override keyword is used appropriately
+// checkOverrides would also be a reasonable name for it.
+static void checkMethodsOverride() {
+
+  TypeToNameToFns map;
+
+  // populate the map
+  forv_Vec(FnSymbol, fn, gFnSymbols) {
```

Comment:
Couldn't you just iterate over gTypeSymbols, then each class' methods?

Reply:
I did it this way because it matches the pattern of other code in this file and so is less likely to find compiler inconsistencies. I expect that all of the definitions of "method" I might want to use aren't always honored by our AST. E.g. fn->_this could be set, whether or not it's in AggregateType->methods.